collector/test/HostInfoTest.cpp

```diff
@@ -114,45 +114,45 @@ TEST(KernelVersionTest, TestVersionStringPopulated) {
   // We're not doing anything with the version string asside from storing it
   // for other things to process, so we can simply test that it is populated
   // correctly in the KernelVersion class
-  KernelVersion version = KernelVersion("5.10.3", "this-is-a-version-string");
+  KernelVersion version = KernelVersion("5.10.3", "this-is-a-version-string", "");
   EXPECT_EQ("this-is-a-version-string", version.version);
 }
 
 TEST(KernelVersionTest, TestHasEBPFSupport) {
-  KernelVersion new_kernel("5.10.0", "");
+  KernelVersion new_kernel("5.10.0", "", "");
   EXPECT_EQ(true, new_kernel.HasEBPFSupport());
 
-  KernelVersion old_kernel("2.6.0", "");
+  KernelVersion old_kernel("2.6.0", "", "");
   EXPECT_EQ(false, old_kernel.HasEBPFSupport());
 
-  KernelVersion old_four_kernel("4.10.0", "");
+  KernelVersion old_four_kernel("4.10.0", "", "");
   EXPECT_EQ(false, old_four_kernel.HasEBPFSupport());
 
-  KernelVersion new_four_kernel("4.20.0", "");
+  KernelVersion new_four_kernel("4.20.0", "", "");
   EXPECT_EQ(true, new_four_kernel.HasEBPFSupport());
 }
 
 TEST(HostInfoTest, TestIsRHEL76) {
-  KernelVersion kernel("3.10.0-957.10.1.el7.x86_64", "");
+  KernelVersion kernel("3.10.0-957.10.1.el7.x86_64", "", "");
   std::string os_id = "rhel";
   EXPECT_TRUE(isRHEL76(kernel, os_id));
 
   os_id = "coreos";
   EXPECT_FALSE(isRHEL76(kernel, os_id));
 
-  kernel = KernelVersion("5.10.0", "");
+  kernel = KernelVersion("5.10.0", "", "");
   EXPECT_FALSE(isRHEL76(kernel, os_id));
 }
 
 TEST(HostInfoTest, TestHasEBPFSupport) {
-  KernelVersion kernel("3.10.0-957.10.1.el7.x86_64", "");
+  KernelVersion kernel("3.10.0-957.10.1.el7.x86_64", "", "");
   std::string os_id = "rhel";
   EXPECT_TRUE(hasEBPFSupport(kernel, os_id));
 
   os_id = "coreos";
   EXPECT_FALSE(hasEBPFSupport(kernel, os_id));
 
-  kernel = KernelVersion("5.10.0", "");
+  kernel = KernelVersion("5.10.0", "", "");
   EXPECT_TRUE(hasEBPFSupport(kernel, os_id));
 }
 
```

Comment:
Please also add unit tests for s390x on 8.4.

Reply:
added tests for HostInfo (for machine id) and HostHeuristics (for 8.4 probe switch)